### PR TITLE
Bluetooth: BR: Add listener cb for discovery

### PR DESCRIPTION
**Pull request by lylezhu2012**
The results of inquiry and extended inquiry are only reported from application level after the inquiry complete event notified.

While the event of inquiry result is notified by controller in real time.

It is not a good user experience.

Just like scanning of LE, add a listener cb for discovery.

When the event of inquiry result, extended inquiry result, or remote name request complete notified, call listener `recv` cb to notify the upper layer.

When the event of inquiry complete notified or no pending of remote name request , call listener `timeout` cb to notify the upper layer.

### CODE REVIEW

File: include/zephyr/bluetooth/bluetooth.h [hunk 1]
```diff
@@ -2577,6 +2580,48 @@ int bt_br_discovery_start(const struct bt_br_discovery_param *param,
  */
 int bt_br_discovery_stop(void);
 
+struct bt_br_discovery_cb {
```
Comment (jhedberg) [inline include/zephyr/bluetooth/bluetooth.h:2583]: All these Classic-related APIs should probably be moved under `include/zephyr/bluetooth/classic/`

Comment (jhedberg) [reply include/zephyr/bluetooth/bluetooth.h:2583]: ..and I think it'd be good to do that before extending the current API

**Comment by jhedberg** [comment]
> Just like scanning of LE, add a listener cb for discovery.

I *think* the reason why we have the two kind of scan callbacks for LE is due to backwards compatibility, i.e. the old singlelton callback was kept as-is when support for multiple independently registered callbacks was added. We don't necessarily have to follow that approach for BR/EDR since the API is not considered stable, i.e. let's do this the way we want to do it as if we were defining the API from scratch. @alwa-nordic @jori-nordic do you guys remember how this went for LE?

That said, I'm all in favor of making the API more asynchronous, i.e. the application should get updates as they happen, rather than the stack withholding the information until it's been able to get everything possible for each discovered device.

File: include/zephyr/bluetooth/bluetooth.h (hunk 1 above)
Comment (lylezhu2012) [reply include/zephyr/bluetooth/bluetooth.h]: Updated

**Comment by lylezhu2012** [comment]
> > Just like scanning of LE, add a listener cb for discovery.
> 
> I _think_ the reason why we have the two kind of scan callbacks for LE is due to backwards compatibility, i.e. the old singlelton callback was kept as-is when support for multiple independently registered callbacks was added. We don't necessarily have to follow that approach for BR/EDR since the API is not considered stable, i.e. let's do this the way we want to do it as if we were defining the API from scratch. @alwa-nordic @jori-nordic do you guys remember how this went for LE?
> 
> That said, I'm all in favor of making the API more asynchronous, i.e. the application should get updates as they happen, rather than the stack withholding the information until it's been able to get everything possible for each discovered device.

Well. I removed the callback parameter from function bt_br_discovery_start.

**Pull request closed without merging**